modules/usbip.nix

```diff
@@ -0,0 +1,58 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  usbipd-win-auto-attach = pkgs.fetchurl {
+    url = "https://raw.githubusercontent.com/dorssel/usbipd-win/9e259e74e225ed9321484366a79545b00a18185c/Usbipd/wsl-scripts/auto-attach.sh";
+    hash = "sha256-KJ0tEuY+hDJbBQtJj8nSNk17FHqdpDWTpy9/DLqUFaM=";
+  };
+
+  cfg = config.wsl.usbip;
+in
+{
+  options.wsl.usbip = with types; {
+    enable = mkEnableOption "USB/IP integration";
+    autoAttach = mkOption {
+      type = listOf str;
+      default = [ ];
+      example = [ "4-1" ];
+      description = "Auto attach devices with provided Bus IDs.";
+    };
+  };
+
+  config = mkIf (config.wsl.enable && cfg.enable) {
+
+    environment.systemPackages = [
+      pkgs.linuxPackages.usbip
+    ];
+
+    services.udev.enable = true;
+
+    systemd = {
+      services."usbip-auto-attach@" = {
+        description = "Auto attach device having busid %i with usbip";
+        after = [ "network.target" ];
+
+        scriptArgs = "%i";
+        path = [
+          pkgs.coreutils
+          pkgs.gnugrep
```

Comment:
Aren't those there by default?

Reply:
You might be correct. I'll try to remove them and see if the script executes.